core/src/main/java/tc/oc/pgm/stats/StatType.java

```diff
@@ -1,43 +1,87 @@
 import static net.kyori.adventure.text.format.NamedTextColor.*;
 import static tc.oc.pgm.stats.StatsMatchModule.damageComponent;
 import static tc.oc.pgm.util.text.NumberComponent.number;
 
 import java.util.Locale;
 import net.kyori.adventure.text.Component;
+import tc.oc.pgm.variables.Variable;
 
-public enum StatType {
-  KILLS,
-  DEATHS,
-  ASSISTS,
-  KILL_STREAK,
-  BEST_KILL_STREAK,
-  KILL_DEATH_RATIO,
-  LONGEST_BOW_SHOT {
-    private final String blocks = key + ".blocks";
+public sealed interface StatType {
+  Component makeNumber(Number number);
+
+  Component buildMessage(Component valueComponent);
+
+  Component conciseComponent(StatHolder stats);
+
+  enum BaseStatType implements StatType {
+    KILLS,
+    DEATHS,
+    ASSISTS,
+    KILL_STREAK,
+    BEST_KILL_STREAK,
+    KILL_DEATH_RATIO,
+    LONGEST_BOW_SHOT {
+      private final String blocks = key + ".blocks";
+
+      @Override
+      public Component makeNumber(Number number) {
+        return translatable(blocks, number(number, YELLOW));
+      }
+    },
+    DAMAGE {
+      @Override
+      public Component makeNumber(Number number) {
+        return damageComponent(number.doubleValue(), GREEN);
+      }
+    };
+
+    public final String key = "match.stats.type." + name().toLowerCase(Locale.ROOT);
 
     @Override
     public Component makeNumber(Number number) {
-      return translatable(blocks, number(number, YELLOW));
+      return number(number, this == DEATHS ? RED : GREEN);
+    }
+
+    @Override
+    public Component conciseComponent(StatHolder stats) {
+      return translatable(key, makeNumber(stats.getStat(this)));
     }
-  },
-  DAMAGE {
+
     @Override
-    public Component makeNumber(Number number) {
-      return damageComponent(number.doubleValue(), GREEN);
+    public Component buildMessage(Component valueComponent) {
+      return translatable(key, valueComponent);
     }
-  };
+  }
 
-  public final String key = "match.stats.type." + name().toLowerCase(Locale.ROOT);
-  public static final StatType[] VALUES = values();
+  final class VariableStatType implements StatType {
+    private static final String SEPERATOR = ":";
 
-  public Component makeNumber(Number number) {
-    return number(number, this == DEATHS ? RED : GREEN);
-  }
+    public final Variable<?> variable;
+    private final Component friendlyName;
 
-  public Component component(StatHolder stats) {
-    return translatable(key, makeNumber(stats.getStat(this)));
+    public VariableStatType(Variable<?> variable, Component friendlyName) {
+      this.variable = variable;
+      this.friendlyName = friendlyName;
+    }
+
+    @Override
+    public Component makeNumber(Number number) {
+      return number(number, GREEN);
+    }
+
+    @Override
+    public Component conciseComponent(StatHolder stats) {
+      return friendlyName.append(
+          text(SEPERATOR + " ").append(makeNumber(stats.getStat(this))).color(WHITE));
```

Comment:
Should probably have a translation key for generic stat instead, so the same way we have `match.stats.type.kills` introduce a `match.stats.type.generic` that defines the format, so like `{0}: {1}`, where one is the stat name and the other the value

Reply:
Seems sensible can do